### ISSUE TITLE
Missing/added characters in a text file after text insertion

### ISSUE THREAD

**Issue by tsavage23** [notepad-plus-plus/notepad-plus-plus#1083]
Missing/added characters in a text file after text insertion

I was working with two text files. I had one loaded in Notepad++ brought up the other in Notepad. I selected part of the second file to paste into the file in Notepad++. I went back to Notepad++ and pasted the text into the top/front of the Notepad++ file, position 0,0, replacing a number of lines of code. At html Validation with W3C Markup Validation Service I had two errors I couldn't resolve. The first Error appeared at position 0,0. The second Error appeared at the junction of the original text and the inserted text. The cause was not apparent since the text appeared to be good. I inspected the file in a hex editor at both positions and found non-printable characters missing at position 0,0 and extra characters at the junction.

Using the W3 Markup Validation Service I retyped the first character at the front of the file and at the junction of the files without effect. I then tried the first two characters at the front of the file and that problem was resolved, and then retyped the two characters at the junction and the second problem was resolved.

It would seem that when inserting text data into a file, at the front of the text, the text is inserted at position 0,0, in front of the three character file code, which then appears within the text file at the front of the remaining original text, resulting in two errors, missing file code and characters in a position not allowed.

The characters are non-printing characters and are thus not visible within Notepad++.  It took significant effort to find the cause when dealing with text files since you can't see the missing/extra non-printing characters.

I hope you can fix it, I find myself using Notepad++ all the time now to edit code (html in this case) and I hope no one else has this problem.

Here is the data I used to resolve the problem:

Error 1: Line 1, Column 1: character "‪" not allowed in prolog
	   <?xml version="1.0" encoding="utf-8" standalone="no"?>

3C 3F 78 6D 6C 20 76 65 72 73 69 6F 6E 3D 22 31 	<?xml version="1
2E 30 22 20 65 6E 63 6F 64 69 6E 67 3D 22 75 74 	.0" encoding="ut
66 2D 38 22 20 73 74 61 6E 64 61 6C 6F 6E 65 3D 	f-8" standalone=
22 6E 6F 22 3F 3E 0D 3C 21 44 4F 43 54 59 50 45 "no"?>.<!DOCTYPE

Fix: Retyped characters 3C 3F (<?) Resutt:
EF BB BF 3C 3F 78 6D 6C 20 76 65 72 73 69 6F 6E 	ï»¿<?xml version
3D 22 31 2E 30 22 20 65 6E 63 6F 64 69 6E 67 3D 	="1.0" encoding=
22 75 74 66 2D 38 22 20 73 74 61 6E 64 61 6C 6F 	"utf-8" standalo
6E 65 3D 22 6E 6F 22 3F 3E 0D 3C 21 44 4F 43 54 	ne="no"?>.<!DOCT

Error 2: Line 83, Column 1: character data is not allowed here

3E 0D 09 09 09 09 3C 2F 74 64 3E 0D 09 09 09 3C 	>.....</td>....<
2F 74 72 3E 0D 09 09 3C 2F 74 61 62 6C 65 3E 0D	/tr>...</table>.
E2 80 AB 0D 20 09 09 3C 70 3E 20 3C 61 20 69 64 	â€«. ..<p> <a id
3D 22 6E 75 6D 5F 63 68 61 70 5F 31 22 20 63 6C 	="num_chap_1" cl
61 73 73 3D 22 63 6E 22 3E 31 3C 2F 61 3E 0D 09		ass="cn">1</a>..

Fix: Retyped characters 0D 0D (leaving 0D 0A 0D 0A) I didn't want to fix it with the Hex Editor.
25 22 20 3E 0D 0A 09 09 09 09 3C 2F 74 64 3E 0D 	%" >......</td>.
0A 09 09 09 3C 2F 74 72 3E 0D 0A 09 09 3C 2F 74 	....</tr>....</t
61 62 6C 65 3E 0D 0A 0D 0A 09 09 3C 68 31 3E D7 	able>......<h1>×
A1 D6 B5 D7 A4 D6 B6 D7 A8 20 D7 91 D6 BC D6 B0 	¡Öµ×¤Ö¶×¨ ×‘Ö¼Ö°
D7 9E D6 B4 D7 93 D6 B0 D7 91 D6 BC D6 B7 D7 A8 	×žÖ´×“Ö°×‘Ö¼Ö•×¨

And got this Message:

    This document was successfully checked as XHTML 1.1!    

If there is anything I can do to help you resolve this problem please let me know.

Tom Savage 

You're missing all the formatting and text illumination ... Oh, Well.

**Comment by dail8859** [comment]
It's hard to follow the exact problem but I'm assuming it comes down to making sure you are using the correct file encoding. The first 3 bytes of `EF BB BF` is due to the [UTF-8 Byte Order Mark](https://en.wikipedia.org/wiki/Byte_order_mark#UTF-8).

**Comment by dail8859** [comment]
Closing due to no response.
